### PR TITLE
ATB-1226 - AIS: Feature-tests Pipeline Config and Docker test execution

### PR DESCRIPTION
**Pull request by ChrisBates1**
### What changed

Update Dockerfile and run-tests.sh to run AIS feature-tests in AWS pipeline

### Why did it change

To allow for testing in Dev and Build through AWS Pipelines

### Issue tracking

- [ATB-1226](https://govukverify.atlassian.net/browse/ATB-1226)

## Testing

Happy Path Test for AIS Submission and Validation successfully locally 
image

## Checklists
- [x] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [x] Tested changes and included test evidence in the PR, if appropriate
- [ ] Included all required tags and other properties for any new resources in the SAM template
- [ ] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [ ] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [x] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [x] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests

[ATB-1226]: https://govukverify.atlassian.net/browse/ATB-1226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

File: feature-tests/Dockerfile [hunk 1]
```diff
@@ -10,19 +10,20 @@ WORKDIR /
...
-	./aws/install
+    ./aws/install
 
 # COPY Files from src directory
 COPY package.json /package.json
 COPY yarn.lock /yarn.lock
 COPY tsconfig.json /tsconfig.json
 COPY jest.config.ts /jest.config.ts
-COPY jest-cucumber-config.js /jest-cucumber-config.js
+# COPY jest-cucumber-config.js /jest-cucumber-config.js
```
Comment (TaherKapasi) [inline feature-tests/Dockerfile:20]: uncomment as the file is referenced in tsconfig.json

**Review by TaherKapasi — DISMISSED** [review]

File: feature-tests/Dockerfile (hunk 1 above)
Comment (ChrisBates1) [reply feature-tests/Dockerfile]: Resolved on latest Commit, thanks Taher

File: feature-tests/run-tests.sh [hunk 2]
```diff
@@ -7,10 +7,10 @@ cp /yarn.lock . 2>/dev/null || :
 cp /tsconfig.json . 2>/dev/null || :
 cp /jest.config.ts . 2>/dev/null || :
 cp /jest-cucumber-config.js . 2>/dev/null || :
-cp -R /tests ./tests 2>/dev/null || :
-cp -R /apiEndpoints ./apiEndpoints 2>/dev/null || :
-cp -R /node_modules ./node_modules 2>/dev/null || :
-cp -R /utils ./utils 2>/dev/null || :
+cp /tests ./tests 2>/dev/null || :
```
[the remark this answers is not in the document]
Comment (ChrisBates1) [reply feature-tests/run-tests.sh:10]: I have added -R back in against lines 11,12,13
however when we use _r on line 10, we get the following error in docker:
image

File: feature-tests/run-tests.sh [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
+cp -R /node_modules . 2>/dev/null || :
+
+cp -R /utils . 2>/dev/null || :
+
 
 # run tests and save the exit code
 declare test_run_result
-yarn test 1>/dev/null
+
+yarn test #1>/dev/null
```
Comment (TaherKapasi) [inline feature-tests/run-tests.sh:29]:
```suggestion
yarn test 1>/dev/null
```

**Review by liamquayle — APPROVED** [review]

**Review by TaherKapasi — APPROVED** [review]

**Review by TaherKapasi — APPROVED** [review]

**Pull request merged**